### PR TITLE
[TFTRT]: Add graphdef visualization

### PR DESCRIPTION
**Pull request by meena-at-work**
Using the environment variable TF_TRT_GRAPH_DUMP_PATH, allow TFTRT to export the optimized TF graphdef in pbtxt format. This should help with debugging.

CC: @DEKHTIARJonathan @pjannaty @bixia1

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc [hunk 1]
```diff
@@ -232,8 +244,15 @@ Status TRTOptimizationPass::Optimize(grappler::Cluster* cluster,
         UpdateFunctionSpecificConversionParams(params_, func_item.func_attr()));
   }
 
-  return ConvertGraph(params_, optimized_item, nodes_to_preserve, cluster,
+  auto status = ConvertGraph(params_, optimized_item, nodes_to_preserve, cluster,
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:247]: We don't use "auto" in this case. see [C++ style guide](https://google.github.io/styleguide/cppguide.html#Type_deduction)

**Review by bixia1 — APPROVED** [review]

File: tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc [hunk 2]
```diff
@@ -232,8 +244,15 @@ Status TRTOptimizationPass::Optimize(grappler::Cluster* cluster,
         UpdateFunctionSpecificConversionParams(params_, func_item.func_attr()));
   }
 
-  return ConvertGraph(params_, optimized_item, nodes_to_preserve, cluster,
+  Status status = ConvertGraph(params_, optimized_item, nodes_to_preserve, cluster,
                       optimized_graph);
+
+  if (!tftrt_graph_dump_path.empty()) {
+    DumpGraphDefToFile(strings::StrCat("after_TRTOptimizer_tf_graph"),
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:251]: Need to use absl::StrCat instead.

**Review by bixia1 — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@meena-at-work Can you please check @bixia1's comments and keep us posted ? Thank you!

**Comment by pjannaty** [comment]
@gbaned please see Meenakshi's change! Cheers!

**Comment by gbaned** [comment]
Hi @bixia1 Can you please review this PR ? Thank you!

**Review by DEKHTIARJonathan — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @meena-at-work Can you please confirm if this PR is still valid? Thank you!

**Pull request closed without merging**